algorithms/src/sorting/impl/Kokkos_SortImpl.hpp

```diff
@@ -305,14 +313,10 @@ void sort_device_view_with_comparator(
     const Experimental::SYCL& exec,
     const Kokkos::View<DataType, Properties...>& view,
     const ComparatorType& comparator) {
-  using ViewType = Kokkos::View<DataType, Properties...>;
-  constexpr bool strided =
-      std::is_same_v<LayoutStride, typename ViewType::array_layout>;
-  if constexpr (strided) {
-    // strided views not supported in dpl so use the most generic case
-    copy_to_host_run_stdsort_copy_back(exec, view, comparator);
-  } else {
+  if (view.stride(0) == 1) {
```

Comment:
We also need to match the checks in `sort_onedpl` above
```
          rank == 1 && 
          (std::is_same_v<typename ViewType::array_layout, LayoutRight> ||
           std::is_same_v<typename ViewType::array_layout, LayoutLeft> ||
           std::is_same_v<typename ViewType::array_layout, LayoutStride>),
```      
here so we can fallback to `copy_to_host_run_stdsort_copy_back` in case of a different layout without running into the `Kokkos::abort` or `static_assert`.

Reply:
ok, so i changed this because it makes sense for this specific function. However, this is within the impl namespace and as far as usability, we have a static assert in the public API that alreay statically checks only for these layouts. so one cannot pass a view with a different layout anyway. makes sense?

Reply:
> we have a static assert in the public API that alreay statically checks only for these layouts.

That's not true for the overloads without comparator, see https://github.com/kokkos/kokkos/blob/d89140de00f90183dc6618167bd30a0f9470970b/algorithms/src/sorting/Kokkos_SortPublicAPI.hpp#L31-L72, but the ones with comparator have them https://github.com/kokkos/kokkos/blob/d89140de00f90183dc6618167bd30a0f9470970b/algorithms/src/sorting/Kokkos_SortPublicAPI.hpp#L77-L131.

Reply:
Then we need this check when calling the version without comparator (since it's also hitting `sort_onedpl`), right?

Reply:
yes, I did not want touch the existing stuff, but makes sense IMO move the constraints as much as possible into the public API rather than having them hidden. So yes i am changing this to also add these constraints to the public API of the sort without comparator.